src/Mod/Draft/WorkingPlane.py

```diff
@@ -694,7 +694,7 @@ def setup(self, direction=None, point=None, upvec=None, force=False):
         if self.weak or force:
             if direction and point:
                 self.alignToPointAndAxis(point, direction, 0, upvec)
-            else:
+            elif FreeCAD.GuiUp:
```

Comment:
Remember what I mentioned in the forum, "is Workingplane supposed to work without the graphical interface or not?" Yorik said it should be non-GUI, so I also think we should try to keep it non-GUI, and avoid graphical functions if possible.

Also, these imports could be done at the beginning of the file. They don't need to be made here.

Reply:
yes, but that Gui part was already there so better to check if Gui is up before running it (also if it is guarded by a try-except function)

Reply:
Indeed, that's why the whole "working plane + snapper grid" system is made in two distinct parts. Maybe we could rearrange that better, have a working plane app (the working plane itself) and a working plane gui (the grid) and at some point add more communication between the two, for ex. the WP could signal the grid when it changes... I'd also like to make the WP document-dependent so you could have several documents opened, or even different views! Each with their own WP+grid

Reply:
I don't know, I didnt want to start working also on this: it will end up in having started a lot of things and having leaved them all at half. So I'm against adding more commits to this PR since my goal is already reached to me.

In the future, yes, I agree we can make several optimizations: we can signal the grid when the WP changes (I'd love that), but also make the WP document dependent. About View depended WP I do not know, since I guess you would lose the possibility to use it when the Gui is not Up... but dont' know... Also there were some posts on the forum to have an unified FreeCAD Grid (and the current Draft grid showed some problems sometimes with transparency and overlays on objects)... Anyway let's see!

Reply:
I agree with Yorik, I think there should be a way to split the GUI functionality of the working plane. It's not a huge priority right now because things works quite well, but it could be done eventually. The working plane can be mathematically defined without a GUI; the GUI would only come into play if there is a GUI and we want to set the working plane by the position of the camera or something that involves the 3D view; this is essentially how it works at the moment I think.

Reply:
I also agree. Just I do not have time to further work on this now, cause I should study it, and I do not know the implications of changing something in a tool that I dont fully understand. :)